powershell_completions.go

```diff
@@ -11,90 +8,253 @@ import (
+    }
+
+    # When completing a flag with an = (e.g., testprog -n=<TAB>)
+    # bash focuses on the part after the =, so we need to remove
+    # the flag part from $cur
+    #if [[ "${cur}" == -*=* ]]; then
+    #    flagPrefix="${cur%%=*}="
+    #    cur="${cur#*=}"
+    #fi
+
+    __%[1]s_debug "Calling $RequestComp"
+    #call the command store the output in $out and redirect stderr and stdout to null
+    # $Out is an array contains each line per element
+    Invoke-Expression -OutVariable out "$RequestComp" 2>&1 | Out-Null
+
+
+    # get directive from last line
+    [int]$Directive = $Out[-1].TrimStart(':')
+    if ($Directive -eq "") {
+        # There is not directive specified
+        $Directive = 0
+    }
+    __%[1]s_debug "The completion directive is: $Directive"
+    
+    # remove directive (last element) from out
+    $Out = $Out | Where-Object { $_ –ne $Out[-1] }
+    __%[1]s_debug "The completions are: $Out"
+
+    if (($Directive -band $ShellCompDirectiveError) -ne 0 ) {
+        # Error code.  No completion.
+        __%[1]s_debug "Received error from custom completion go code"
+        return
+    }
+
+    $Longest = 0
+    $Values = $Out | ForEach-Object {
+        #Split the output in name and description
+`+"        $Name, $Description = $_.Split(\"`t\",2)"+`
+        __%[1]s_debug "Name: $Name Description: $Description"
+
+        # Look for the longest completion so that we can format things nicely
+        if ($Longest -lt $Name.Length) {
+            $Longest = $Name.Length
+        }
+
+        # Set the description to a one space string if there is none set.
+        # This is needed because the CompletionResult does not accept an empty string as argument
+        if (-Not $Description) {
+            $Description = " "
+        }
+        @{Name="$Name";Description="$Description"}
+    }
+
+
+    $Space = " "
+    if (($Directive -band $ShellCompDirectiveNoSpace) -ne 0 ) {
+        # remove the space here
+        __%[1]s_debug "ShellCompDirectiveNoSpace is called"
+        $Space = ""
+    }
+
+    if (($Directive -band $ShellCompDirectiveNoFileComp) -ne 0 ) {
+        __%[1]s_debug "ShellCompDirectiveNoFileComp is called"
+        
+        if ($Values.Length -eq 0) {
+            # Just print an empty string here so the 
+            # shell does not start to complete paths.
+            # We cannot use CompletionResult here because 
+            # it does not accept an empty string as argument.
+            ""
+            return
+        }
+    }
+
+    if (($Directive -band $ShellCompDirectiveFilterFileExt) -ne 0 ) {
 
-func escapeStringForPowerShell(s string) string {
-	return strings.Replace(s, "'", "''", -1)
+        ## get the paths from Get-ChildItem
+        __%[1]s_debug "ShellCompDirectiveFilterFileExt"
+
+        $Values | ForEach-Object {
+            __%[1]s_debug "Filter: $($_.Name)"
+            Get-ChildItem -Recurse -Name -File -Filter "*.$($_.Name)"
```

Comment:
This seems to work pretty decently.  I thought you said `ShellCompDirectiveFilterFileExt` wasn't working?

Also, why to you recurse into sub directories?

Reply:
Yeah it works kinda, but I can't make it handle paths correctly. I can only make it complete the full path or only the file name. It won't work with relative paths. It also breaks completely with spaces.

That's why I'm thinking that sticking to standard path completion done by powershell would be better. This is just a ugly hack and doesn't seam to be supported by powershell in any way.